### PR TITLE
Prepare modules ec2_vpc_egress_igw for promotion

### PR DESCRIPTION
**Pull request by abikouo**
##### SUMMARY

- Refactor module `ec2_vpc_egress_igw` to use shared code from `amazon.aws.plugins.module_utils.ec2`
- ec2_vpc_egress_igw - add support for tagging

##### ISSUE TYPE

- Feature Pull Request
- New Module Pull Request

##### COMPONENT NAME

`ec2_vpc_egress_igw`

### CODE REVIEW

File: plugins/modules/ec2_vpc_egress_igw_info.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      - A dict of filters to apply. Each dict item consists of a filter key and a filter value.
+        See U(https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeEgressOnlyInternetGateways.html) for possible filters.
+    type: dict
+    default: {}
+  egress_only_internet_gateway_ids:
+    description:
+      - Get details of specific VPC Egress Internet Gateway ID.
+    type: list
+    elements: str
+  convert_tags:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_vpc_egress_igw_info.py:27]: Isn't this conversion something we want done by default?

File: plugins/modules/ec2_vpc_egress_igw_info.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  - amazon.aws.common.modules
+  - amazon.aws.region.modules
+  - amazon.aws.boto3
+"""
+
+EXAMPLES = r"""
+# # Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: Gather information about all VPC Egress only Internet Gateways for an account or profile
+  amazon.aws.ec2_vpc_igw_info:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_vpc_egress_igw_info.py:42]: It's still community.aws

File: plugins/modules/ec2_vpc_egress_igw_info.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        attachments:
+            description: Information about the attachment of the egress-only internet gateway.
+            returned: always
+            type: complex
+            contains:
+                state:
+                    description: The current state of the attachment.
+                    returned: always
+                    type: str
+                    sample: available
```
Comment (alinabuzachis) [inline plugins/modules/ec2_vpc_egress_igw_info.py:75]:
```suggestion
                    sample: "available"
```

File: plugins/modules/ec2_vpc_egress_igw_info.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                vpc_id:
+                    description: The ID of the VPC.
+                    returned: always
+                    type: str
+                    sample: vpc-02123b67
```
Comment (alinabuzachis) [inline plugins/modules/ec2_vpc_egress_igw_info.py:80]:
```suggestion
                    sample: "vpc-02123b67"
```

File: plugins/modules/ec2_vpc_egress_igw_info.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        egress_only_internet_gateway_id:
+            description: The ID of the egress-only internet gateway.
+            returned: always
+            type: str
+            sample: eigw-0123456789abcdef
```
Comment (alinabuzachis) [inline plugins/modules/ec2_vpc_egress_igw_info.py:85]:
```suggestion
            sample: "eigw-0123456789abcdef"
```

File: changelogs/fragments/20240923-refactor-ec2_vpc_egress_igw-modules.yml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+---
+minor_changes:
+  - ec2_vpc_egress_igw - Refactor module to use shared code from ``amazon.aws.plugins.module_utils.ec2`` util ().
+  - ec2_vpc_egress_igw - Add the possibility to update/add tags on Egress only internet gateway ().
```
Comment (alinabuzachis) [inline changelogs/fragments/20240923-refactor-ec2_vpc_egress_igw-modules.yml:4]:
```suggestion
  - ec2_vpc_egress_igw - Refactor module to use shared code from ``amazon.aws.plugins.module_utils.ec2`` util (https://github.com/ansible-collections/community.aws/pull/2152).
  - ec2_vpc_egress_igw - Add the possibility to update/add tags on Egress only internet gateway (https://github.com/ansible-collections/community.aws/pull/2152).
```

File: plugins/modules/ec2_vpc_egress_igw_info.py [hunk 7]
```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+---
+module: ec2_vpc_egress_igw_info
```
Comment (alinabuzachis) [inline plugins/modules/ec2_vpc_egress_igw_info.py:9]: I would probably suggest adding this new module directly into amazon.aws rather than here.

Comment (abikouo) [reply plugins/modules/ec2_vpc_egress_igw_info.py]: @alinabuzachis the new module has been removed and will be introduced with the migration of `ec2_vpc_egress_igw` to `amazon.aws`

**Review by alinabuzachis — APPROVED** [review]

**Review by GomathiselviS — APPROVED** [review]

**Pull request merged**